### PR TITLE
Add a modal to force user to specify its convention

### PR DESCRIPTION
**Pull request by desoindx**

### CODE REVIEW

File: controllers/configController.ts [hunk 1]
```diff
@@ -11,11 +24,12 @@ function getConfig(req: Request, res: Response): void {
     dateOfBirthDeploymentDate: config.dateOfBirthDeploymentDate,
     sessionDuration: config.sessionDurationHours,
     satistics: config.satistics,
+    universities,
```
Comment (sandrica89) [inline controllers/configController.ts:27]: Why add it in config and not call it in ConventionForm the only place where it is used?

File: controllers/loginController.ts [hunk 2]
```diff
@@ -69,6 +69,9 @@ async function saveToken(email: string, token: string) {
 
 const connectedUser = async (req: Request, res: Response): Promise<void> => {
   const psy = await dbPsychologists.getPsychologistById(req.user.psychologist);
+  const psychologistId = req.user.psychologist;
```
Comment (sandrica89) [inline controllers/loginController.ts:72]:
```suggestion
  const psychologistId = req.user.psychologist;
  const psy = await dbPsychologists.getPsychologistById(psychologistId);
```

File: frontend/cypress/integration/utils/login.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const loginAsDefault = (duration = 3600) => {
 };
 
 const logout = () => {
-  cy.get('[data-test-id="logout-button"]').click();
+  cy.get('[data-test-id="logout-button"]').click({ force: true });
```
Comment (sandrica89) [inline frontend/cypress/integration/utils/login.js:25]: Why did you add `force: true` here?

File: frontend/src/components/Psychologist/Appointments/ConventionModal.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import { observer } from 'mobx-react';
+import { Modal, ModalTitle, ModalContent } from '@dataesr/react-dsfr';
+import { useStore } from 'stores/';
+import { checkConvention } from 'services/conventionVerification';
+import ConventionForm from '../Reimbursement/ConventionForm';
+
+const ConventionModal = ({ currentConvention }) => {
+  const { commonStore: { config } } = useStore();
+  const [isOpen, setIsOpen] = useState(true);
```
Comment (sandrica89) [inline frontend/src/components/Psychologist/Appointments/ConventionModal.jsx:10]: Is `isOpen` needed, since this component is not render when conditions doesn't match (see Appointment.jsx l.67-71)?

File: test/demarcheSimplifiees/autoVerifyPsychologist.spec.js [hunk 5]
```diff
@@ -71,5 +71,5 @@ describe('autoVerifyPsychologist', () => {
           instructeurId: INSTRUCTOR_ID,
         },
       }));
-  }).timeout(4000);
+  }).timeout(30000);
```
Comment (sandrica89) [inline test/demarcheSimplifiees/autoVerifyPsychologist.spec.js:74]: :o that's a lot.. should we ask demarches simplifiees about why is it that long the first time?

File: controllers/reimbursementController.ts [hunk 6]
```diff
@@ -71,9 +56,11 @@ const updateConventionInfo = async (req: Request, res: Response): Promise<void>
 
   const psychologistId = req.user.psychologist;
   await dbPsychologists.updateConventionInfo(psychologistId, universityId, isConventionSigned);
+  const convention = await dbPsychologists.getConventionInfo(psychologistId);
```
[the remark this answers is not in the document]
Comment (sandrica89) [reply controllers/reimbursementController.ts:59]: Also, we could keep it as it was here, and on front side, instead of setting the new convention info in the user store, simply call a `pullUser`.

**Comment by sandrica89** [comment]
In `AppointmentController.getAppointments`, we retrieve the convention info. I don't think this is needed anymore since we have them in the userStore.

**Comment by sandrica89** [comment]
When the popin show up, the mailto is focused by default. I would expect it to be focus only if I hit TAB key. (This is not blocking from my POV)
popin_convention

**Review by sandrica89 — CHANGES_REQUESTED** [review]
See PR comments

File: controllers/configController.ts (hunk 1 above)
Comment (desoindx) [reply controllers/configController.ts:27]: because i was too lazy to make a new endpoint :(

File: controllers/reimbursementController.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (desoindx) [reply controllers/reimbursementController.ts:59]: not easily... because the convention object is a join between two tables :/ so not only what we update

File: frontend/cypress/integration/utils/login.js (hunk 3 above)
Comment (desoindx) [reply frontend/cypress/integration/utils/login.js:25]: because it can be hidden by the convention popup

File: frontend/src/components/Psychologist/Appointments/ConventionModal.jsx (hunk 4 above)
Comment (desoindx) [reply frontend/src/components/Psychologist/Appointments/ConventionModal.jsx:10]: it's smoother to close it manually because the convention can take time to update :/
so not really needed but a bit better

File: test/demarcheSimplifiees/autoVerifyPsychologist.spec.js (hunk 5 above)
Comment (desoindx) [reply test/demarcheSimplifiees/autoVerifyPsychologist.spec.js:74]: i wanted to make sure we will never be bother by it :)

**Review by sandrica89 — APPROVED** [review]
